great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -426,6 +495,9 @@ def _create_batch_spec_kwargs(self) -> dict[str, Any]:
         }
 
 
+SQLAsset: TypeAlias = Union[_SQLAsset, QueryAsset, TableAsset]
```

Comment:
I don't think we should be doing this.
If we don't like the way `_SQLAsset` looks we can change the mechanism the `Metadatasource` uses to register types and then we can `_SQLAsset` -> `SQLAsset`.

In my head, it would just be a `_register_type: ClassVar[bool] = True/Flase` attribute.
I can create a ticket for that.

Reply:
Per our offline conversation I have reverted this `TypeAlias` change.